### PR TITLE
ddl, session: after upgrades, the order of processing on different tables is also more as expected

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/44304

Problem Summary:

### What is changed and how it works?
If the job's `IsPausedBySystem` is true, we need to return this get a job(with the paused state) to get a new job(with queuing state).

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/job_table.go [hunk 1]
```diff
@@ -203,7 +203,7 @@ func (d *ddl) processJobDuringUpgrade(sess *sess.Session, job *model.Job) (isRun
 	if job.IsPausedBySystem() && !hasSysDB(job) {
 		var errs []error
 		errs, err = ResumeJobsBySystem(sess.Session(), []int64{job.ID})
-		if len(errs) > 0 {
+		if len(errs) > 0 && errs[0] != nil {
```
Comment (tangenta) [inline ddl/job_table.go:206]: Is it possible to have `errs` like `[nil, "some err", nil]`?

Comment (zimulala) [reply ddl/job_table.go:206]: Yes.

Comment (zimulala) [reply ddl/job_table.go:206]: @tangenta PTAL

**Review by tangenta — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**